src/lib/executeSourceTransfer.js

```diff
@@ -2,6 +2,12 @@
 
 const log = require('../common').log.create('executeSourceTransfer')
 const validator = require('./validate')
+const promiseRetry = require('promise-retry')
+const retryOpts = { // retry 10 times within 5 seconds
+  retries: 10,
+  factor: 1,
+  minTimeout: 500
+}
```

Comment:
I think it would be better to start with 10 ms and double the wait each time it fails until the transfer is expired (basically setting `retries` to Infinity and putting the promise retry in a [race](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/race) with a timeout that will trigger when the transfer is expired)